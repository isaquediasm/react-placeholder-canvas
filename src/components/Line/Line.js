import React from 'react';
import PropTypes from 'prop-types';
import { invertIntervals, renderIntervals } from '../../helpers';
import { StyledOffset, StyledLine } from './Line.styled';

const Component = ({ height, intervals }) => (
  <StyledLine height={height}>
    {renderIntervals(invertIntervals(intervals)).map((styling, idx) => (
      <StyledOffset key={idx} style={styling} />
    ))}
  </StyledLine>
);

const Line = React.memo(Component);

Line.propTypes = {
  height: PropTypes.number.isRequired,
  intervals: PropTypes.array,
};

export default Line;
