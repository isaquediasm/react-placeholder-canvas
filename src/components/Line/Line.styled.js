import styled from 'styled-components';

const StyledOffset = styled.div`
  position: absolute;
  display: inline-block;
  background-color: #fff;
  height: inherit;
`;

const StyledLine = styled.div`
  height: ${props => props.height}px;
`;

export { StyledOffset, StyledLine };
