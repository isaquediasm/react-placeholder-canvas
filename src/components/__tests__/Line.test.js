import React from 'react';
import renderer from 'react-test-renderer';
import Line from '../Line/Line';

it('should render lines correctly', () => {
  const tree = renderer
    .create(<Line height={10} intervals={[[0, 10], [20, 30]]} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
