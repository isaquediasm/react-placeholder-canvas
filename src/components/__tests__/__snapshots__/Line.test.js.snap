// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render lines correctly 1`] = `
<div
  className="sc-1hs1toa-1 dXlBMd"
  height={10}
>
  <div
    className="sc-1hs1toa-0 ekigly"
    style={
      Object {
        "left": "10%",
        "width": "10%",
      }
    }
  />
  <div
    className="sc-1hs1toa-0 ekigly"
    style={
      Object {
        "left": "30%",
        "width": "70%",
      }
    }
  />
</div>
`;
